### ISSUE TITLE
Feature request: zip analog with failed promises

### ISSUE THREAD

**Issue by numen31337** [malcommac/Hydra#33]
I think this will be helpful to have some function analog to the 'zip' function which will accept the list of promises, wait until all of them is finished and returns both failed and resolved in the result.

**Comment by r-plus** [comment]
In that case I think you should wrap the resolved value and error to one object (e.g. Result).

**Comment by numen31337** [comment]
Yes, that's clear, but in this case, it forces me to use some other way of error handling in addition to this library.
